### ISSUE TITLE
SSL test has no SSL-related calls

### ISSUE THREAD

**Issue by olegabr** [pintsized/lua-resty-http#108]
Look at this file please: https://github.com/pintsized/lua-resty-http/blob/master/t/09-ssl.t

There are no SSL-related functions called. No `https` urls queried. Am I missing something?

**Comment by olegabr** [comment]
I'm warring about it because I've tried to use this lib to load urls form facebook servers, that are all with `https` protocol, and failed with cryptic messages in logs. I've searched for an example of SSL functionality usage here, but found nothing. Now I see that it is not really tested.
No examples, no tests, my attempt to use it failed. I have no clue what I'm doing wrong, and moreover, I'm not sure if it is supposed to work with `https` at all.

**Comment by pintsized** [comment]
Yes, it's problematic to do within the test harness. I'd like to improve this (and the docs), but haven't had the time. Patches are more than welcome.

I can tell you that this library serves huge amounts of SSL traffic on probably thousands of sites, so yes it works. It works because `ngx_lua` provides this functionality for us, and certainly is well tested. That is, the underlying code for actual SSL work is not within this library. Although I agree, some end-to-end tests should exist here.

If you have an actual problem, feel free to raise an issue with details or perhaps ask questions on the OpenResty mailing list.

The error logs are typically not cryptic, perhaps you just don't understand them. Paste them here and we'll try to help you.
